satellite/orders/endpoint.go

```diff
@@ -167,55 +189,31 @@ func (endpoint *Endpoint) Settlement(stream pb.Orders_SettlementServer) (err err
 			continue
 		}
 
-		bucketID, err := endpoint.DB.UseSerialNumber(ctx, orderLimit.SerialNumber, orderLimit.StorageNodeId)
-		if err != nil {
-			log.Warn("unable to use serial number", zap.Error(err))
-			if ErrUsingSerialNumber.Has(err) {
-				err := monitoredSettlementStreamSend(ctx, stream, &pb.SettlementResponse{
-					SerialNumber: orderLimit.SerialNumber,
-					Status:       pb.SettlementResponse_REJECTED,
-				})
-				if err != nil {
-					return formatError(err)
-				}
-				continue
-			}
-			return err
-		}
-		now := time.Now().UTC()
-		intervalStart := time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), 0, 0, 0, now.Location())
+		requests = append(requests, &ProcessOrderRequest{Order: order, OrderLimit: orderLimit})
 
-		projectID, bucketName, err := SplitBucketID(bucketID)
-		if err != nil {
-			return err
-		}
-		if err := endpoint.DB.UpdateBucketBandwidthSettle(ctx, *projectID, bucketName, orderLimit.Action, order.Amount, intervalStart); err != nil {
-			// TODO: we should use the serial number in the same transaction we settle the bandwidth? that way we don't need this undo in an error case?
-			if err := endpoint.DB.UnuseSerialNumber(ctx, orderLimit.SerialNumber, orderLimit.StorageNodeId); err != nil {
-				log.Error("unable to unuse serial number", zap.Error(err))
+		if len(requests) >= endpoint.settlementBatchSize {
```

Comment:
This would end up deadlocking with the client if it sends orders one at a time.

Reply:
Should the defer at 127 cover this?
```
	defer func() {
		if len(requests) > 0 {
			err = errs.Combine(err, endpoint.processOrders(ctx, stream, requests))
			if err != nil {
				err = formatError(err)
			}
		}
	}()
```

Reply:
Not really the main thing is let's say the storage node implements the settlement as:

```
for _, order := range orders {
    send order
    wait for response
}
```

Over time we would expect completely new implementations of storage nodes... in which case the simplest implementation of writing that settlement is that code. However, we can require the async implementation to be the protocol.

Reply:
so no change for now?

Reply:
I would either:
1. fix the system such that it supports that specific implementation
2. document that this implementation requires async implementation

Reply:
Creating a new PR for #2